src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -26,6 +27,7 @@ internal sealed class NamedPipeConnectionListener : IConnectionListener
     private readonly PipeOptions _inputOptions;
     private readonly PipeOptions _outputOptions;
     private readonly Mutex _mutex;
+    private readonly ConcurrentQueue<NamedPipeServerStream> _streamsCache = new ConcurrentQueue<NamedPipeServerStream>();
```

Comment:
(no action needed)

out of idle curiosity, I thought I'd bench "bag" (`TryTake`/`Add`) vs "queue" (`TryDequeue`/`Enqueue`) vs "stack" (`TryPop`/`Push`) here (in the "there's something to take" case):

```
| Method |     Mean |    Error |   StdDev |   Gen0 | Allocated |
|------- |---------:|---------:|---------:|-------:|----------:|
|  Queue | 18.17 ns | 0.164 ns | 0.145 ns |      - |         - |
|  Stack | 22.45 ns | 0.130 ns | 0.116 ns | 0.0038 |      32 B |
|    Bag | 61.40 ns | 0.560 ns | 0.524 ns |      - |         - |
```

it just makes me wonder: what's the point of "bag"? (`</rant>`)

Reply:
> what's the point of "bag"?

Unordered and having a thread-local and a global list (to allow stealing). 
But I think it needs some optimizations...not done sinde TPL got added?